sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -22,137 +22,64 @@
-        Map<String, List<Expression>> settings = new HashMap<>();
-        Assignment<Expression> assignment;
+        Map<Symbol, List<Symbol>> settings = new HashMap<>();
+
+        var exprAnalyzer = new ExpressionAnalyzer(
+            functions,
+            txnCtx,
+            typeHints,
+            FieldProvider.FIELDS_AS_LITERAL,
+            null
+        );
+
+        SetStatement<Symbol> statement = node.map(x -> exprAnalyzer.convert(x, new ExpressionAnalysisContext()));
 
         switch (node.scope()) {
             case LICENSE:
-                if (node.assignments().size() != SetLicenseAnalyzedStatement.LICENSE_TOKEN_NUM) {
+                if (node.assignments().size() != AnalyzedSetLicenseStatement.LICENSE_TOKEN_NUM) {
                     throw new IllegalArgumentException("Invalid number of arguments for SET LICENSE. " +
                                                        "Please provide only the license key");
                 }
-                assignment = node.assignments().get(0);
-                String licenseKey = ExpressionToStringVisitor.convert(assignment.expression(), Row.EMPTY);
-
-                return new SetLicenseAnalyzedStatement(licenseKey);
+                return new AnalyzedSetLicenseStatement(statement.assignments().get(0).expression());
             case GLOBAL:
-                for (Assignment<Expression> anAssignment : node.assignments()) {
-                    for (String setting : ExpressionToSettingNameListVisitor.convert(anAssignment)) {
-                        CrateSettings.checkIfRuntimeSetting(setting);
-                    }
-                    String settingName = ExpressionToStringVisitor.convert(anAssignment.columnName(), Row.EMPTY);
-                    settings.put(settingName, ImmutableList.of(anAssignment.expression()));
+                for (Assignment<Symbol> a : statement.assignments()) {
+                    Symbol settingName = a.columnName();
+                    settings.put(settingName, ImmutableList.of(a.expression()));
                 }
                 break;
             default:
-                assignment = node.assignments().get(0);
-                // parser does not allow using the parameter expressions as setting names in the SET statements,
-                // therefore it is fine to convert the expression to string here.
-                String settingName = ExpressionToStringVisitor.convert(assignment.columnName(), Row.EMPTY);
-                List<String> nameParts = CrateSettings.settingNamesByPrefix(settingName);
-                if (nameParts.size() != 0) {
-                    throw new IllegalArgumentException(String.format(Locale.ENGLISH,
-                        "GLOBAL Cluster setting '%s' cannot be used with SET SESSION / LOCAL", settingName));
-                }
+                Assignment<Symbol> assignment = statement.assignments().get(0);
+                Symbol settingName = assignment.columnName();
```

Comment:
this looks the same as at the ``GLOBAL`` branch, so the ``GLOBAL`` impl can be default.